### PR TITLE
align first add of launch configuration to preferences

### LINKED ISSUE #5634: Add debug configuration for the first time is hard coded to use .theia folder
### Description
when you add debug configuration for the first time it also ignores the preference (.vscode and .theia folders): https://github.com/theia-ide/theia/blob/af00d42b3d71b0d0b723cb970fefafb9c8500281/packages/debug/src/browser/debug-configuration-manager.ts#L255

### Reproduction Steps
1. Clean .vscode or .theia folder from the workspace if there is any
1. Override preference configuration to getPaths return only .vscode
1. Run command add debug configuration
1. The configuration is created anyway in .theia folder due to the hard coded name in the code above

**OS and Theia version:** All

**Diagnostics:** 
N/A

### Question:
If I want to fix it what is the correct API to call? Is it PreferenceConfigurations.getPaths()[0] or is there any abstraction on top of it that I should use?

### PR DESCRIPTION
**Pull request by amiramw**
fix wrong behavior of creating it always hard coded under .theia folder

Fixes #5634

### CODE REVIEW

File: packages/debug/src/browser/debug-configuration-manager.ts [hunk 1]
```diff
@@ -280,14 +271,11 @@ export class DebugConfigurationManager {
...
-    protected getInitialConfigurationContent(initialConfigurations: DebugConfiguration[]): string {
-        return `{
-  // Use IntelliSense to learn about possible attributes.
-  // Hover to view descriptions of existing attributes.
-  "version": "0.2.0",
-  "configurations": ${JSON.stringify(initialConfigurations, undefined, '  ').split('\n').map(line => '  ' + line).join('\n').trim()}
-}
-`;
+    protected getInitialConfigurationContent(initialConfigurations: DebugConfiguration[]): {version: string, configurations: DebugConfiguration[]} {
+        return {
```
Comment (akosyakov) [inline packages/debug/src/browser/debug-configuration-manager.ts]: Can we preserve comments?

For example set `{}` as `launch`, resolve `configUri` after that and then write to this URI as before?

File: packages/debug/src/browser/debug-configuration-manager.ts [hunk 2]
```diff
@@ -258,18 +254,13 @@ export class DebugConfigurationManager {
...
         const content = this.getInitialConfigurationContent(configurations);
-        try {
-            await this.filesystem.createFile(uri.toString(), { content });
-        } catch (e) {
-            if (!FileSystemError.FileExists.is(e)) {
-                throw e;
-            }
-        }
-        return uri;
+        await this.preferences.set('launch', content);
```
Comment (akosyakov) [inline packages/debug/src/browser/debug-configuration-manager.ts]: generally good idea!

File: packages/debug/src/browser/debug-configuration-manager.ts [hunk 3]
```diff
@@ -258,18 +254,13 @@ export class DebugConfigurationManager {
...
+        const { configUri } = this.preferences.resolve('launch');
+        // @ts-ignore
```
Comment (vince-fugnitto) [inline packages/debug/src/browser/debug-configuration-manager.ts]: Is the `// @ts-ignore` necessary?
The variable `configUri` may be undefined which does not respect the return type of the method.
Should we handle this more gracefully than suppressing the warning?

File: packages/debug/src/browser/debug-configuration-manager.ts (hunk 1 above)
Comment (amiramw) [reply packages/debug/src/browser/debug-configuration-manager.ts]: now comments are preserved. but i think it makes the code more complex.

File: packages/debug/src/browser/debug-configuration-manager.ts (hunk 3 above)
Comment (amiramw) [reply packages/debug/src/browser/debug-configuration-manager.ts]: now there is a fallback for this case

File: packages/debug/src/browser/debug-configuration-manager.ts [hunk 4]
```diff
@@ -258,12 +258,23 @@ export class DebugConfigurationManager {
...
     }
     protected async doCreate(model: DebugConfigurationModel): Promise<URI> {
-        const uri = new URI(model.workspaceFolderUri).resolve('.theia/launch.json');
+        await this.preferences.set('launch', {}); // create dummy launch.json in the correct place
+        const { configUri } = this.preferences.resolve('launch'); // get uri to write content to it
+        let uri: URI;
+        if (configUri && configUri.path.base === 'launch.json') {
+            uri = configUri;
+        } else { // fallback
+            uri = new URI(model.workspaceFolderUri).resolve('.theia/launch.json');
```
Comment (akosyakov) [inline packages/debug/src/browser/debug-configuration-manager.ts]: we should use the first config path here, not `.theia`, i mean from `preferenceconfigurations.parths`

Comment (amiramw) [reply packages/debug/src/browser/debug-configuration-manager.ts]: fixed

**Pull request merged**